.github/workflows/main.yml

```diff
@@ -0,0 +1,47 @@
+name: run-test
+
+# Controls when the workflow will run
+on:
+  push:
+  pull_request:
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+    # Test the code on multiple python versions
+    strategy:
+      matrix:
+        python-version: ["3.7", "3.8", "3.9", "3.10", "3.11"]
+    env:
+      working-directory: 
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v3
+
+      # Setup python and install requirements - specified manually atm!
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+          architecture: 'x64'
+
+      # Install requirements - specified manually atm!
+      - name: Install dependencies
+        run: |
+          python3 -m pip install --upgrade pip
+          pip install numpy
+          pip install pandas
+
+      # Runs a single command using the runners shell
+      - name: Run a one-line script
+        run: echo Hello, world!
+
+      # Runs a set of commands using the runners shell
+      - name: Run a multi-line script
```

Comment:
Probably want to update `name` here before it's integrated into CI pipeline. Just so that you have consistency in the future. Same story with lines 42 and 43.

Reply:
you mean something more informative and specific to this particular test, correct?

Reply:
Yeah that's right. Something like "Run unit tests"